.github/workflows/ci.yaml

```diff
@@ -18,30 +23,38 @@ jobs:
     runs-on: ${{ matrix.OS }}-latest
     name: "tests"
     strategy:
+      fail-fast: false
       matrix:
-        PYTHON:
-          - 3.8
-          - 3.9
-        OS:
-          - ubuntu
-          - windows
-          - macos
+        python-version: [3.8, 3.9]
+        os: ['ubuntu', 'windows', 'macos']
+
     steps:
       - uses: actions/checkout@v2
+
       - uses: conda-incubator/setup-miniconda@v2
-        auto-update-conda: true
-        python-version: ${{ matrix.PYTHON }}
-        miniforge-variant: Mambaforge
-      - name: "Temporary installation (reqs and code)"
-        run: python -m pip install .[test]
-        # the above step should be changed to only install reqs
-      - name: "Versions"
-        run: conda list
+        with:
+            auto-update-conda: true
+            use-mamba: true
+            python-version: ${{ matrix.python-version }}
+            miniforge-variant: Mambaforge
+            environment-file: environment.yml
+            activate-environment: openfe
+
       - name: "Install"
-        run: python -m pip install --no-deps .
-      - name: "Check imports"
+        run: python setup.py develop --no-deps
+
+      - name: "Environment Information"
+        run: |
+          mamba info -a
+          mamba list
+
+      - name: "Run tests"
         run: |
-          python -c "import openfe.setup"
-          python -c "import openfe.orchestration"
-          python -c "import openfe.simulation"
-          python -c "import openfe.analysis"
```

Comment:
I see your point, although my follow up is: do we have a reasonable case where we could face the issue of failing to import any of the sub-modules if importing `openfe` works fine?

I realise we might be doing namespace stuff here, but at the same time we might want to test this package in a modular way, so testing for all the imports here might not be super clean long term?

Reply:
Currently set up as a namespace package, so importing `openfe` doesn't actually do anything meaningful.

I added `import foo` to `setup/__init__.py`:

```python
In [1]: import openfe

In [2]: import openfe.setup
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Input In [2], in <module>
----> 1 import openfe.setup

File ~/Dropbox/omsf/src/openfe/openfe/setup/__init__.py:1, in <module>
----> 1 import foo

ModuleNotFoundError: No module named 'foo'
```

Reply:
> edit: essentially my question is to scope out how much we're testing the install here, do we want to keep it just to these four components going forward, or do we want a smaller/bigger subset, etc...

This comes back to the question of whether we split the repo. If we want to go with namespace packages, my vote is to split the repo and to do so sooner rather than later. Each repo would test its own import. If we don't want namespace packages, then we change this to be a normal package and you'd only need to test `import openfe` here.

I guess I don't find namespace packages so confusing, and I really like the additional modularity they allow. But deciding on that is clearly going to affect how we set things up here.